java/com/google/copybara/git/GitHubPrDestination.java

```diff
@@ -337,14 +337,18 @@ private String getPullRequestBranchName(
       return gitHubDestinationOptions.destinationPrBranch;
     }
     String contextReference = changeRevision.contextReference();
-    // We could do more magic here with the change identity. But this is already complex so we
-    // require  a group identity either provided by the origin or the workflow (Will be implemented
-    // later.
-    checkCondition(contextReference != null,
-        "git.github_pr_destination is incompatible with the current origin. Origin has to be"
-            + " able to provide the contextReference or use '%s' flag",
-        GitHubDestinationOptions.GITHUB_DESTINATION_PR_BRANCH);
-    String branchNameFromUser = getCustomBranchName(contextReference);
+
+    String branchNameFromUser = prBranch;
```

Comment:
I think the code is a bit confusing after this change.

When prBranch == null, we get branchNameFromUser == null (now the only reason to enter the if statement),

But then we call:

```  private String getCustomBranchName(String contextReference) throws ValidationException {
    if (prBranch == null) {
      return null;
    }
...
```

effectively returning null.

I would refactor this code to have something that makes sense overall.

Also, could you add a tests in GitHubPrDestinationTest?

Thanks for the contribution!!

Reply:
Addressed changes in https://github.com/google/copybara/pull/229